### PR TITLE
Add a management command to update from a PopoloSource

### PR DESCRIPTION
**Pull request by mhl**
You can specify the PopoloSource to update either by URL or ID. If that
PopoloSource doesn't exist, you'll get an error and a list of those that
do exist. If you specify --create and a URL, it'll add the source and
then update it.

### CODE REVIEW

File: popolo_sources/management/commands/popolo_sources_update.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return PopoloSource.objects.create(url=source_arg)
+
+
+class Command(BaseCommand):
+
+    help = 'Update from source of Popolo JSON'
+
+    def add_arguments(self, parser):
+        parser.add_argument(REQUIRED_ARG)
+        parser.add_argument('--create', action='store_true', default=False)
```
Comment (dracos) [inline popolo_sources/management/commands/popolo_sources_update.py]: I believe setting action=store_true automatically sets a default of False.

**Review by dracos — CHANGES_REQUESTED** [review]
Some tiny potential changes, otherwise 🥚

File: popolo_sources/management/commands/popolo_sources_update.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    try:
+        split_url = urlsplit(s)
+    except ValueError:
+        return False
+    return bool(split_url.scheme)
+
+
+def get_source(source_arg):
+    if re.search(r'^\d+$', source_arg):
+        source_id = int(source_arg, 10)
```
Comment (mhl) [inline popolo_sources/management/commands/popolo_sources_update.py]: Oh, it turns out I was under a misapprehension about how `int` behaved - I thought that if you didn't specify a base then `010` was interpreted as octal, but actually `int('010')` is `10`. I'll take that out :)

File: popolo_sources/management/commands/popolo_sources_update.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                raise CommandError(msg)
+        except PopoloSource.DoesNotExist:
+            if create:
+                ps = create_source(source_arg)
+                print("Created a source for that URL: {0}".format(ps.id))
+            else:
+                print("That source could not be found.")
+                if PopoloSource.objects.exists():
+                    print("Did you mean one of the following?")
+                    for existing_source in PopoloSource.objects.order_by('pk'):
```
Comment (mhl) [inline popolo_sources/management/commands/popolo_sources_update.py]: I suppose so, if I had a clear idea whether `id` or `url` would be more useful generally as the default order! I think I'd rather leave it unspecified at the moment - specifying a default order would mean having to check for the effect on write-it, for example.

**Pull request merged**